### PR TITLE
CB-11740: Integrate with Telemetry solution.

### PR DESCRIPTION
**Pull request by juhi-09**
CB-11740: Integrate with Telemetry solution
Adding a new property to CDPEnvironmentAzureDetails, bool resourceEncryptionEnabled, this is set only when encryptionKeyUrl is present.
As encryptionKeyUrl is present only when "CDP_CB_AZURE_DISK_SSE_WITH_CMK" is enabled and encryptionKey is provided during environment creation, checking its presence is sufficient to know if user is using "Azure SSE with CMK".

**Thunderhead changes** - https://github.infra.cloudera.com/thunderhead/thunderhead/pull/5132

**When encryptionKeyUrl was provided :**
Screenshot 2021-06-24 at 4 42 21 PM

**When encryptionKeyUrl was NOT provided :**
Screenshot 2021-06-24 at 4 53 03 PM

### CODE REVIEW

**Review by nadamdb — CHANGES_REQUESTED** [review]
Please add the descriptive comment in usage.proto, otherwise LGTM

File: structuredevent-service-cdp/src/main/java/com/sequenceiq/cloudbreak/structuredevent/service/telemetry/converter/EnvironmentDetailsToCDPEnvironmentDetailsConverter.java [hunk 1]
```diff
@@ -96,6 +98,12 @@
             if (azureParametersDto != null) {
                 builder.setSingleResourceGroup(
                         azureParametersDto.getAzureResourceGroupDto().getResourceGroupUsagePattern().isSingleResourceGroup());
+                Optional<String> encryptionKeyUrl = Optional.of(azureParametersDto)
+                        .map(AzureParametersDto::getAzureResourceEncryptionParametersDto)
+                        .map(AzureResourceEncryptionParametersDto::getEncryptionKeyUrl);
+                if (encryptionKeyUrl.isPresent()) {
```
[the remark this answers is not in the document]
Comment (lajosrodek) [reply structuredevent-service-cdp/src/main/java/com/sequenceiq/cloudbreak/structuredevent/service/telemetry/converter/EnvironmentDetailsToCDPEnvironmentDetailsConverter.java:104]: +1

File: usage-collection/src/main/proto/usage.proto [hunk 2]
```diff
@@ -760,6 +760,7 @@ message CDPFreeIPADetails {
 // Azure specific environment details
 message CDPEnvironmentAzureDetails {
   bool singleResourceGroup = 1;
+  bool resourceEncryptionEnabled = 2;
```
[the remark this answers is not in the document]
Comment (lajosrodek) [reply usage-collection/src/main/proto/usage.proto:763]: +1

**Review by lajosrodek — APPROVED** [review]
I agree with Adam's comments, please fix them. Otherwise looks okay.

File: usage-collection/src/main/proto/usage.proto [hunk 3]
```diff
@@ -760,6 +760,8 @@ message CDPFreeIPADetails {
 // Azure specific environment details
 message CDPEnvironmentAzureDetails {
   bool singleResourceGroup = 1;
+  // True if user provides encryption key url to encrypt Azure disk with CMK
```
Comment (lajosrodek) [inline usage-collection/src/main/proto/usage.proto:763]: Nit:

- url -> URL
- disk -> managed disks

**Review by nadamdb — APPROVED** [review]

**Comment by lajosrodek** [comment]
As per Bryan's comment in the [thunderhead PR](https://github.infra.cloudera.com/thunderhead/thunderhead/pull/5132):
> While you are at it would you mind also adding a comment above the `singleResourceGroup` field?

So one more change needed, please. 😄

**Pull request merged**